host/src/lib/HelperFunctions.jsx

```diff
@@ -318,23 +319,18 @@ export const checkEqualityWithPrevAnswer = (normValue, prevAnswerValue, prevAnsw
  * @param {IResponse[]} prevAnswer
  * @returns {boolean}
  */
-export const checkEqualityWithOtherAnswers = (rawAnswer, normValue, normType, prevAnswer) => {
-  // loop through each of the normalized answers in each of the previous answers
-  for (let i = 0; i < prevAnswer.normAnswer.length; i++) {
-      if (normType === prevAnswer.normAnswer[i].type && checkEqualityWithPrevAnswer(normValue, prevAnswer.normAnswer[i].value, prevAnswer.normAnswer[i].type)) {
-        return true;
-      }
-      // last ditch check on expressions in case there is a basic match that the type check in the previous conditional misses
-      if ((normType === AnswerType.EXPRESSION || prevAnswer.normAnswer[i].type === AnswerType.EXPRESSION) 
-        && (normValue.toString() === prevAnswer.normAnswer[i].value.toString())) {
-          return true;
-      }
-      // last ditch raw answer checks (5% === 5% enter as an expression, for instance)
-      if (rawAnswer === prevAnswer.value) {
-        return true;
-      }
-  };
-  return false;
+export const checkEqualityWithOtherAnswers = (rawAnswer, normAnswerType, normAnswer,  prevAnswer, correctAnswerRegex) => {
```

Comment:
trying to optimize this as much as possible by using a set to handle the nested loops. I don't think there's a way around it for the expressions (as we need to insert that `mathjs` function to compare, so moving that off to its own function and returning from this one early if we use it